pkg/vulnsrc/rocky/rocky.go

```diff
@@ -143,22 +139,45 @@ func (vs *VulnSrc) put(errataVer map[string][]RLSA) error {
 }
 
 func (vs *VulnSrc) commit(tx *bolt.Tx, platformName string, errata []RLSA) error {
+	savedInputs := map[string]PutInput{}
 	for _, erratum := range errata {
 		for _, cveID := range erratum.CveIDs {
-			advisories := map[string]types.Advisory{}
+			input := PutInput{
+				Advisories: map[string]types.Advisory{},
+			}
+			if in, ok := savedInputs[cveID]; ok {
+				input = in
+			}
 			for _, pkg := range erratum.Packages {
 				// Skip the modular packages until the following bug is fixed.
 				// https://forums.rockylinux.org/t/some-errata-missing-in-comparison-with-rhel-and-almalinux/3843/8
 				if strings.Contains(pkg.Release, ".module+el") {
 					continue
 				}
 
-				advisories[pkg.Name] = types.Advisory{
+				fixedVersion := types.FixedVersion{
 					FixedVersion: utils.ConstructVersion(pkg.Epoch, pkg.Version, pkg.Release),
+					Arch:         pkg.Arch,
+					VendorID:     erratum.ID,
+				}
+
+				// if the advisory for this package and CVE have been kept - just add the new architecture
+				if adv, ok := input.Advisories[pkg.Name]; ok {
+					// don't include duplicates
+					if !adv.FixedVersions.IsDuplicate(fixedVersion) {
```

Comment:
I guess `slices.Contains` works.

```suggestion
					if !slices.Contains(adv.FixedVersions, fixedVersion) {
```

Reply:
Good idea! 
Changed in https://github.com/aquasecurity/trivy-db/pull/321/commits/14b1e69623e02e77ffd327040011f50da3997643